src/main/webapp/js/instructorFeedbackEdit/visibilityOptions.js

```diff
@@ -142,41 +152,79 @@ function disableRow($containingForm, row) {
     $tdElements.parent().hide();
 }
 
-function updateEditTabAccordingToRecipient($containingForm) {
+function disableRowsAccordingToRecipient($containingForm) {
     var recipientType = $containingForm.find('select[name="recipienttype"]').val();
-    if (isRecipientsTeamMembersVisibilityOptionInvalidForRecipientType(recipientType)) {
-        // show the row Recipient(s) and hide the row Recipient's Team Members
-        enableRow($containingForm, ROW_RECIPIENT);
+    switch (recipientType) {
+    case 'SELF':
+        // ROW_RECIPIENT is disabled because self-feedback is always visible to giver
+        disableRow($containingForm, ROW_RECIPIENT);
+        // ROW_RECIPIENT_TEAM is disabled because it is the same as ROW_GIVER_TEAM
         disableRow($containingForm, ROW_RECIPIENT_TEAM);
-        return;
-    } else if (recipientType === 'NONE') {
-        // hide both the row Recipient(s) and the row Recipient's Team Members
+        break;
+    case 'STUDENTS':
+        // all options enabled when recipientType is STUDENTS (subject to options disabled by giverType)
+        break;
+    case 'OWN_TEAM':
+        // ROW_RECIPIENT is disabled because it is the same as ROW_GIVER_TEAM
+        disableRow($containingForm, ROW_RECIPIENT);
+        // ROW_RECIPIENT_TEAM is disabled because it is same as ROW_RECIPIENT
```

Comment:
Actually I think it is better to keep `ROW_GIVER_TEAM` because since giver == recipient, we should not introduce the idea of a separate `ROW_RECIPIENT`. E.g., if you create a student self-feedback question, you will think whether the giver can see, and not whether the recipient can see.